soup.go

```diff
@@ -72,24 +120,22 @@ func GetWithClient(url string, client *http.Client) (string, error) {
 		if debug {
 			panic("Couldn't perform GET request to " + url)
 		}
-		return "", errors.New("couldn't perform GET request to " + url)
+		return "", newError(ErrInGetRequest, "couldn't perform GET request to "+url)
 	}
 	defer resp.Body.Close()
 	bytes, err := ioutil.ReadAll(resp.Body)
 	if err != nil {
 		if debug {
 			panic("Unable to read the response body")
 		}
-		return "", errors.New("unable to read the response body")
+		return "", newError(ErrReadingResponse, "unable to read the response body")
 	}
 	return string(bytes), nil
 }
 
 // Get returns the HTML returned by the url in string using the default HTTP client
 func Get(url string) (string, error) {
-	// Init a new HTTP client
-	client := &http.Client{}
```

Comment:
We can initialize this earlier and save some resources if someone is using repeated calls to `soup.Get`

Reply:
Sure.